packages/terra-theme-provider/src/ThemeProvider.jsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const ThemeProviderThemes = {
+  CONSUMER: 'cerner-consumer-theme',
+  MOCK: 'cerner-mock-theme',
+};
+
+const propTypes = {
+  /**
+   * The component(s) that will be wrapped by `<ThemeProvider />`
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Name of class for specified theme. e.g `cerner-consumer-theme`
+   */
+  themeName: PropTypes.string,
+};
+
+const ThemeProvider = ({ themeName, children, ...customProps }) => (
+  <div {...customProps} className={themeName}>{children}</div>
+);
+
+ThemeProvider.propTypes = propTypes;
+ThemeProvider.Opts = {};
+ThemeProvider.Opts.Themes = ThemeProviderThemes;
```

Comment:
I am thinking about two designs of sub-theme handling:
1. Multi-class selector
```
@include terra-consumer-theme {
    &.terra-consumer-theme-concept1 {
        --terra-consumer-color: #123456;
    }
     --terra-button-active-background-color-primary: darken(var(--terra-consumer-color, #1c1f21), 50%);
    ....
}

const theme = `${ThemeProvider.Opts.Themes.CONSUMER} terra-consumer-theme-concept1`
<ThemeProvider themeName={theme}>
```

2. Nested class selector
```
@include terra-consumer-theme {
    .terra-consumer-theme-concept1 {
        --terra-consumer-color: #123456;
    }
     --terra-button-active-background-color-primary: darken(var(--terra-consumer-color, #1c1f21), 50%);
    ....
}

<ThemeProvider themeName={ThemeProvider.Opts.Themes.CONSUMER}>
    <ThemeProvider themeName="terra-consumer-theme-concept1" >
        ...
        <Button />
    <ThemeProvider/>
<ThemeProvider/>
```
@bjankord @mhemesath any thought?